startup_runners/outer_startup_runner.c

```diff
@@ -6,28 +6,45 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <errno.h>
 #include <stdint.h>
 #include <unistd.h>
 
 #include "libkruntime.h"
 
 #define BENCH_FUNC_NAME "run_iter"
+#define LEN_DATUM 8
 
 int
 main(int argc, char **argv)
 {
     double start_time = -1;
     int result;
+    char fake_datum[LEN_DATUM] = "[0, 0], ";
```

Comment:
I think there's a buffer overrun here: `LEN_DATUM` is length 8, but the RHS string is of length 9 (with the null terminator). I think this probably impacts other parts of the code too.